tests/Cache/MemoryCacheItemPoolTest.php

```diff
@@ -0,0 +1,189 @@
+ * limitations under the License.
+ */
+
+namespace Google\Auth\Tests;
+
+use Google\Auth\Cache\Item;
+use Google\Auth\Cache\MemoryCacheItemPool;
+
+class MemoryCacheItemPoolTest extends \PHPUnit_Framework_TestCase
+{
+    private $pool;
+
+    public function setUp()
+    {
+        $this->pool = new MemoryCacheItemPool();
+    }
+
+    public function saveItem($key, $value)
+    {
+        $item = $this->pool->getItem($key);
+        $item->set($value);
+        $this->assertTrue($this->pool->save($item));
+
+        return $item;
+    }
+
+    public function testGetsFreshItem()
+    {
+        $item = $this->pool->getItem('item');
+        
+        $this->assertInstanceOf(Item::class, $item);
+        $this->assertNull($item->get());
+        $this->assertFalse($item->isHit());
+    }
+
+    public function testGetsExistingItem()
+    {
+        $key = 'item';
+        $value = 'value';
+        $this->saveItem($key, $value);
+        $item = $this->pool->getItem($key);
+
+        $this->assertInstanceOf(Item::class, $item);
+        $this->assertEquals($value, $item->get());
+        $this->assertTrue($item->isHit());
+    }
+
+    public function testGetsMultipleItems()
+    {
+        $keys = ['item1', 'item2'];
+        $items = $this->pool->getItems($keys);
+
+        $this->assertEquals($keys, array_keys($items));
+        $this->assertContainsOnlyInstancesOf(Item::class, $items);
+    }
+
+    public function testHasItem()
+    {
+        $existsKey = 'does-exist';
+        $this->saveItem($existsKey, 'value');
+
+        $this->assertTrue($this->pool->hasItem($existsKey));
+        $this->assertFalse($this->pool->hasItem('does-not-exist'));
+    }
+
+    public function testClear()
+    {
+        $key = 'item';
+        $this->saveItem($key, 'value');
+
+        $this->assertTrue($this->pool->hasItem($key));
+        $this->assertTrue($this->pool->clear());
+        $this->assertFalse($this->pool->hasItem($key));
+    }
+
+    public function testDeletesItem()
+    {
+        $key = 'item';
+        $this->saveItem($key, 'value');
+        
+        $this->assertTrue($this->pool->deleteItem($key));
+        $this->assertFalse($this->pool->hasItem($key));
+    }
+
+    public function testDeletesItems()
+    {
+        $keys = ['item1', 'item2'];
+
+        foreach ($keys as $key) {
+            $this->saveItem($key, 'value');
+        }
+
+        $this->assertTrue($this->pool->deleteItems($keys));
+        $this->assertFalse($this->pool->hasItem($keys[0]));
+        $this->assertFalse($this->pool->hasItem($keys[1]));
+    }
+
+    /**
+     * @expectedException \Psr\Cache\InvalidArgumentException
+     */
+    public function testDoesNotDeleteItemsWithInvalidKey()
+    {
+        $keys = ['item1', '{item2}'];
+
+        foreach ($keys as $key) {
+            $this->saveItem($key, 'value');
+        }
+
+        $this->pool->deleteItems($keys);
+
+        $this->assertTrue($this->pool->hasItem($keys[0]));
```

Comment:
Looks like this test will never reach this assert. We should remove the `@expectedException` and put `deleteItems` in a `try/catch` block.